percy/render/recipe_parser.py

```diff
@@ -0,0 +1,1208 @@
+        """
+        Given a value supported by JSON, use the RecipeParser to generate a list
+        of child nodes. This effectively creates a new subtree that can be
+        used to patch other parse trees.
+        """
+        # Multiline values can replace the list of children with a single
+        # multiline leaf node.
+        if isinstance(value, str) and "\n" in value:
+            return [
+                _Node(
+                    value=value.splitlines(),
+                    multiline_flag=True,
+                )
+            ]
+
+        # For complex types, generate the YAML equivalent and build
+        # a new tree.
+        if isinstance(value, (dict, list)):
+            return RecipeParser(  # pylint: disable=W0212
+                yaml.dump(value)
+            )._root.children
+
+        # Primitives can be safely stringified to generate a parse tree.
+        return RecipeParser(  # pylint: disable=W0212
+            str(RecipeParser._stringify_yaml(value))
+        )._root.children
+
+    def _rebuild_selectors(self) -> None:
+        """
+        Re-builds the selector look-up table. This table allows quick access to
+        tree nodes that have a selector specified.
+
+        This needs to be called when the three is modified.
+        """
+        self._selector_tbl: dict[str, list[SelectorInfo]] = {}
+        selector_re = re.compile(r"\[.*\]")
+
+        def _collect_selectors(node: _Node, path: _StrStackImmutable):
+            # Ignore empty comments
+            if not node.comment:
+                return
+            match = selector_re.search(node.comment)
+            if match:
+                selector = match.group(0)
+                selector_info = SelectorInfo(node, path)
+                if selector not in self._selector_tbl:
+                    self._selector_tbl[selector] = [selector_info]
+                else:
+                    self._selector_tbl[selector].append(selector_info)
+
+        _Traverse.traverse_all(self._root, _collect_selectors)
+
+    def __init__(self, content: str):
+        """
+        Constructs a RecipeParser instance.
+        :param content: conda-build formatted recipe file, as a single text
+                        string.
+        """
+        # The initial, raw, text is preserved for diffing and debugging purposes
+        self._init_content: Final[str] = content
+        # Indicates if the original content has changed
+        self._is_modified = False
+
+        # Tracks Jinja variables set by the file
+        self._vars_tbl: dict[str, JsonType] = {}
+        # Find all the set statements and record the values
+        set_line_re = re.compile(r"{%\s*set.*=.*%}\s*\n")
+        for line in set_line_re.findall(self._init_content):
+            key = line[line.find("set") + len("set") : line.find("=")].strip()
+            value = line[line.find("=") + len("=") : line.find("%}")].strip()
+            self._vars_tbl[key] = ast.literal_eval(value)
```

Comment:
```suggestion
            try:
                self._vars_tbl[key] = ast.literal_eval(value)
            except:
                self._vars_tbl[key] = value
```
Would support any value and would still give you a python type other than string for simple values.